_kbarticles/2024-12-06-migrating-opensearch.md

```diff
@@ -0,0 +1,129 @@
+
+**If you used the same saved object in Kibana across multiple Cloud.gov orgs**,
+you will need to import it into each OpenSearch tenant (each tenant corresponds to a Cloud.gov Cloud Foundry "org").
+
+Once you've exported the objects as `.ndjson` files, switch to OpenSearch, then:
+
+* From the LeftNav menu, select, "Management -> Dashboards Management".
+* On the "Dashboards Management" window, select "Saved Objects" on the left menu.
+* Then select "Import" on the upper right corner.
+* On the "Import saved objects" window:
+  * Select the file to upload
+  * For "Import options" select:
+    * "Check for existing objects"
+    * "Request action on conflict"
+  * The click the "Import" button, as show below: ![Screenshot of Import saved objects, with selections as described]({{site.baseurl}}/assets/images/content/opensearch-import-saved-objects.png)
+  * If the import results in an "Overwrite index-pattern" dialog, you will likely want to "Skip" the overwrite: ![Screenshot of Overwrite index-pattern with "Skip" selected]({{site.baseurl}}/assets/images/content/opensearch-import-overwrite-dialog.png)
+  * When the import is complete, click "Done"
+
+
+
+### Recovering Saved Searches and Visualizations after Kibana decomissioning
+
+If you missed migrating a Saved Object
+from Kibana to OpenSearch, and Kibana is no longer available,
+please contact [Cloud.gov Support](mailto:support@cloud.gov).
+We have saved all customer objects and can recover those for you.
+
+## User interface changes
+
+The screenshot below show some of the major changes to the user interfaces, such as:
+1. The drop down menus have moved from the upper left to the upper right
+2. The "Top 5 values" for a field view is now an option to the right of the field, instead of a double-click
+3. There are a lot more values gathered for container metrics
+
+![Screenshot comparing Kibana to OpenSearch]({{site.baseurl}}/assets/images/content/opensearch-ui-differences.png)
+
+## Key system differences
+
+The Cloud.gov team has implemented OpenSearch to deliver a number of benefits to our customers. Among these are:
+
+* Twelve months of live access to system logs, in alignment with [M-21-31](https://www.whitehouse.gov/wp-content/uploads/2021/08/M-21-31-Improving-the-Federal-Governments-Investigative-and-Remediation-Capabilities-Related-to-Cybersecurity-Incidents.pdf).
+* Definitions of saved searches and visualizations are now isolated by OpenSearch tenants that correspond to Cloud.gov organizations.
+  * You no longer need to worry about choosing a globally unique name.
+  * If you share the same saved object across multiple orgs, you will need to import it into each of your orgs.
+* Better handling of large log messages. Both Kibana/ELK and OpenSearch have a 32kb limit on message size. The older system dropped such messages from Kibana (although they were still retained in cold storage), the newer system keeps the first 32kb and discards the rest
```

Comment:
Currently we're only truncating JSON messages that are over 32KB, but we probably should do it for all messages